extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java

```diff
@@ -303,14 +368,16 @@ private PreGeneratedProxies generatedProxies(Set<String> entityClassNames, Index
 
             final BytecodeProviderImpl bytecodeProvider = new BytecodeProviderImpl();
             final ByteBuddyProxyHelper byteBuddyProxyHelper = bytecodeProvider.getByteBuddyProxyHelper();
+            final EntitiesClassLoader classLoader = new EntitiesClassLoader(transformedEntities,
+                    Thread.currentThread().getContextClassLoader());
 
             for (String entity : entityClassNames) {
                 Set<Class<?>> proxyInterfaces = new HashSet<>();
                 proxyInterfaces.add(HibernateProxy.class); //always added
-                Class<?> mappedClass = Class.forName(entity, false, Thread.currentThread().getContextClassLoader());
```

Comment:
Isn't it enough to delay generating the proxies until after the classes are transformed? Isn't the TCCL at this point the QuarkusClassLoader which will load the transformed classes?